File: 3412-permutation-difference-between-two-strings/3412-permutation-difference-between-two-strings.cpp
class Solution {
public:
    int findPermutationDifference(string s, string t) {
        unordered_map<char, int> table;
        int result = 0;

        for (int i = 0; i < s.size(); i++) {
            table[s[i]] = i;
        }

        for (int i = 0; i < t.size(); i++) {
            if (table.count(t[i])) {
                result += abs(table[t[i]] - i);
            }
        }

        return result;
    }
};

File: 3412-permutation-difference-between-two-strings/3412-permutation-difference-between-two-strings.java
class Solution {
    public int findPermutationDifference(String s, String t) {
        int result = 0;

        for (int i = 0; i < s.length(); i++) {
            for (int j = 0; i < t.length(); j++) {
                if (s.charAt(i) == t.charAt(j)) {
                    result += Math.abs(i - j);
                    break;
                }
            }
        }

        return result;
    }
}